### ISSUE TITLE
New gtags integration shadows key for find-symbol in elisp

### ISSUE THREAD

**Issue by jbeigel** [syl20bnr/spacemacs#6619]
#### Description :octocat:
In commit b22702bc9e ggtags and helm-gtags were added to many language layers.

Unfortunately for emacs-lisp, SPC m g g previously worked very well (I think
it was using  elisp-slime-nav), but is now shadowed by helm-gtags-dwim.

Of course the workaround would be to not use the gtags layer, but for other
languages it is fine.

#### Reproduction guide :beetle:
- Start Emacs
- Open some elisp file

*Observed behaviour:* :eyes: :broken_heart:
- Position cursor over some symbol and press SPC m g g => helm-gtags asks
  if it should create a gtags database.
- There is no keybinding for the old "find-symbol" method.

*Expected behaviour:* :heart: :smile:
- Being able to navigate elisp without using gtags

#### System Info :computer:
- OS: gnu/linux
- Emacs: 24.5.1
- Spacemacs: 0.105.21
- Spacemacs branch: develop (rev. 9773ad6)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(markdown auto-completion
          (c-c   :variables c-c  -default-mode-for-headers 'c  -mode)
          emacs-lisp git gtags happymake haskell helm my-magit-gerrit
          (org :variables org-enable-reveal-js-support t)
          osx python
          (rtags :variables rtags-layer-enable-auto-completion t rtags-layer-enable-syntax-checking t)
          shell
          (syntax-checking :variables syntax-checking-enable-tooltips nil)
          version-control yaml)
```

#### Backtrace :paw_prints:
```
<<BACKTRACE IF RELEVANT>>
```

**Comment by TheBB** [comment]
Can we make a list of languages with better `,gg` so we can remove that binding in those languages?

As a workaround for now you can use `(gtags :disabled-for emacs-lisp)`

**Comment by TheBB** [comment]
Relevant: https://github.com/syl20bnr/spacemacs/pull/6620

**Comment by deb0ch** [comment]
Additionally, ggtags conflicts with helm-gtags as it overrides the `M-.` in c++-mode.

Which brings another question: why do we even need them both at the same time ?

Would it not be more reasonable to have something like 
```
(setq c-c++-packages
  '(
    cc-mode
    disaster
    clang-format
    cmake-mode
    company
    (company-c-headers :toggle (configuration-layer/package-usedp 'company))
    company-ycmd
    flycheck
    gdb-mi
    ggtags :toggle (not (configuration-layer/package-usedp 'helm))
    helm-cscope
    helm-gtags :toggle (configuration-layer/package-usedp 'helm)
    semantic
    srefactor
    stickyfunc-enhance
    ycmd
    xcscope
    ))
```
in every language ?

**Comment by tuhdo** [comment]
`eldoc` is such a nice feature that even a package solely for it is ok:

![ggtags_with_asm](https://cloud.githubusercontent.com/assets/4818719/16985349/f79d2a88-4ea9-11e6-8181-130e4c22c321.png)

You can think of `ggtags` in Spacemacs context as `gtags-eldoc`, with extra features aside from just being `eldoc`.

**Comment by deb0ch** [comment]
To try.

In c++-mode if both `semantic` and `c-c++` layers are activated, will it be ggtags-eldoc or semantic taking care of the echo area ?

I found out recently that semantic eldoc-like feature was quite buggy in cpp files, maybe will it be better ?

Moreover, semantic eldoc-like feature had that annoying side-effect of covering all echo area messages, most likely by redisplaying the same "eldoc hints" every second or sth like that, even when you don't move your cursor.

I wonder if it will be the case with gtags-eldoc.

**Comment by TheBB** [comment]
It will either be semantic, or semantic + ggtags advice. I'm not 100% sure what the advice does, if it completely takes over or if it does something more intelligent.

**Comment by TheBB** [comment]
Actually I just had a look, the ggtags advice is an `:after-until` type, so it only takes effect if the original function returns nil. So presumably you will only get ggtags eldoc in those cases where semantic doesn't come up with anything.

**Comment by TheBB** [comment]
Maybe we should add-hook ggtags always at the end so that it never takes over eldoc entirely on its own.

**Comment by tuhdo** [comment]
> I found out recently that semantic eldoc-like feature was quite buggy in cpp files, maybe will it be better ?

What is buggy about it?

**Comment by deb0ch** [comment]
sometimes it echoes an error message instead of the advice

**Comment by TheBB** [comment]
Should avoid binding `,gg` in go as well, see https://github.com/syl20bnr/spacemacs/issues/6663

**Comment by TheBB** [comment]
https://github.com/syl20bnr/spacemacs/pull/6815 is the newest attempt to bring jumping into a harmonious whole.

**Comment by TheBB** [comment]
Merged #6851 which should fix this.
